presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -733,13 +738,17 @@ public void onFailure(Throwable throwable)
             }, directExecutor());
         }
         else {
-            transitionToFinished();
+            transitionToFinished(null);
```

Comment:
we don't pass nulls; it's too dangerous. It's like we pass nullptr in C++

Reply:
Then what to pass then? How about a magic number?

Reply:
Use Optional.empty() as the parameter.